WooCommerce/Classes/POS/Presentation/ItemListView.swift

```diff
@@ -133,6 +133,20 @@ private extension ItemListView {
             }
             .padding(.bottom, floatingControlAreaSize.height)
             .padding(.horizontal, Constants.itemListPadding)
+            .background(GeometryReader { proxy in
+                Color.clear
+                    .onChange(of: proxy.frame(in: .global).maxY) { maxY in
+                        if viewModel.state == .loading {
+                            return
+                        }
+                        let viewHeight = UIScreen.main.bounds.height
+                        if maxY < viewHeight {
+                            Task {
+                                await viewModel.populatePointOfSaleItems()
+                            }
+                        }
+                    }
+            })
```

Comment:
non-blocking: maybe this could be extracted to a modifier to detect scrolling to a threshold?

Reply:
It can! It's on the list [here](https://github.com/woocommerce/woocommerce-ios/issues/14186)